### PR TITLE
Add tests to cover basic functionality

### LINKED ISSUE #16: Tests that cover the base functionality
Add tests that cover the basic functionality

### PR DESCRIPTION
**Pull request by matt-fidd**
Fixes #16

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+name: Test
+
+on:
+  push:
+    branches:
+      - master
```
Comment (twk3) [inline .github/workflows/test.yml:6]:
```suggestion
      - main
```

I used main for the default branch

**Review by twk3 — CHANGES_REQUESTED** [review]
@matt-fidd one nitpick.  Amazing work!

File: .github/workflows/test.yml (hunk 1 above)
Comment (matt-fidd) [reply .github/workflows/test.yml]: Good catch, thank you. Turns out it was an issue on all of the workflows so updated now.

**Review by twk3 — APPROVED** [review]
Awesome! Thanks @matt-fidd

**Pull request merged**